.github/workflows/main.yml

```diff
@@ -100,9 +103,31 @@ jobs:
         cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} -vv
         cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} --release -vv
         cargo run --release --target ${{ matrix.target }} --manifest-path testcrate/Cargo.toml --features package -vv
+        cargo clean --manifest-path testcrate/Cargo.toml
+        rm -f Cargo.lock
       if: startsWith(matrix.os, 'windows')
       name: Run tests (Windows)
 
+    - run: |
+        curl -O https://www.nasm.us/pub/nasm/releasebuilds/${{ env.WINNASMVERSION }}/win32/nasm-${{ env.WINNASMVERSION }}-win32.zip
+        unzip nasm-${{ env.WINNASMVERSION }}-win32.zip
+        export PATH=$PWD/nasm-${{ env.WINNASMVERSION }}:$PATH
+        cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} -vv
+        cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} --release -vv
+        cargo run --release --target ${{ matrix.target }} --manifest-path testcrate/Cargo.toml --features package -vv
+      if: startsWith(matrix.os, 'windows') && startsWith(matrix.target, 'i686')
+      name: Run tests with nasm (Windows i686)
+
+    - run: |
+        curl -O https://www.nasm.us/pub/nasm/releasebuilds/${{ env.WINNASMVERSION }}/win64/nasm-${{ env.WINNASMVERSION }}-win64.zip
+        unzip nasm-${{ env.WINNASMVERSION }}-win64.zip
+        export PATH=$PWD/nasm-${{ env.WINNASMVERSION }}:$PATH
+        cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} -vv
+        cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} --release -vv
+        cargo run --release --target ${{ matrix.target }} --manifest-path testcrate/Cargo.toml --features package -vv
+      if: startsWith(matrix.os, 'windows') && startsWith(matrix.target, 'x86_64')
+      name: Run tests with nasm (Windows x86_64)
```

Comment:
Instead of being subsequent steps could these be two new matrix entries to allow them to run in parallel as well?